umake/frameworks/ide.py

```diff
@@ -139,23 +148,76 @@ def get_sha_and_start_download(self, download_result):
             logger.error("Download page changed its syntax or is not parsable (missing sha512)")
             UI.return_main_screen(status_code=1)
         logger.debug("Found download link for {}, checksum: {}".format(url, sha512))
-        self.download_requests.append(DownloadItem(url, Checksum(self.checksum_type, sha512)))
+        self.download_requests.append(DownloadItem(url, Checksum(ChecksumType.sha512, sha512)))
         self.start_download_and_install()
 
     def post_install(self):
         """Create the Eclipse launcher"""
-        icon_filename = "icon.xpm"
-        icon_path = join(self.install_path, icon_filename)
+        DownloadCenter(urls=[DownloadItem(self.icon_url, None)],
+                       on_done=self.save_icon, download=True)
```

Comment:
That could be done as the same time that you download the framework itself, line 151, isn't it? If you don't want to override the normal "done" callback for extraction and such, you can just move that line itself under line 151.

Also, that fixes the fact that the launcher icon should be created once the icon is saved at the right place, desktop file refreshement can be racy and won't show up until next reboot.